### ISSUE TITLE
Rendering using equirectangular and gaussian splats is not working

### ISSUE THREAD

**Issue by alexandrukis** [nerfstudio-project/nerfstudio#2679]
Hi!

First of all thank you for all the great work! Using nerfstudio is a joy, and it is very helfpul to the community in my opinion.

Second of all, I am not sure if this is a bug or _feature request_, but here we go.

**Describe the bug**
Hi, I am trying to render a scene using an equirectangular camera on a gaussian splatting trained model.

I have used the same rendering method using other methods (nerfacto, nerfacto-big) and I had no issues whatsoever.

This is the camera path I am using:

```json
{
    "keyframes": [],
    "camera_type": "equirectangular",
    "render_height": 2060,
    "render_width": 3840,
    "camera_path": [
        {
            "camera_to_world": [
                0.13874818242650622,
                0.09883540684852354,
                0.9853834300547315,
                0,
                0.978637578226042,
                0.13874818242650622,
                -0.15171497077483276,
                0,
                -0.15171497077483276,
                0.9853834300547315,
                -0.07747298507456568,
                0,
                0,
                0,
                0,
                1
            ],
            "fov": 50,
            "aspect": 1
        }
    ],
    "fps": 1,
    "seconds": 1,
    "smoothness_value": 0.5,
    "is_cycle": false,
    "crop": null
}
```

And this is the command I am running:

```shell
ns-render camera-path --load-config <config_path> --camera-path-filename <camera_path> --output-path <output_path> --output-format images
```

**To Reproduce**

1. Train a model using gaussian splatting
2. Try to render an image using a camera path

**Expected behavior**

I would expect an equirectangular image to be rendered, such as (this images is created by rendering a 360 image of a nerfacto model):

<img width="1673" alt="image" src="https://github.com/nerfstudio-project/nerfstudio/assets/4930102/9bd38238-2fcf-4f63-88c2-1f8121e91245">

**Actual behavior**

A regular, perspective camera is rendered:

<img width="1678" alt="image" src="https://github.com/nerfstudio-project/nerfstudio/assets/4930102/b20e9179-cf81-4ab9-b0c0-b4577b6c4c37">

**Comment by brentyi** [comment]
Yeah, I can confirm that this is just not implemented. Some work would need to happen in `get_outputs()` here to support non-perspective cameras: https://github.com/nerfstudio-project/nerfstudio/blob/ddedb920dc7c2d5cd4be4a042f9718f5bff22981/nerfstudio/models/gaussian_splatting.py#L519C35-L528

This feels like it could be tricky; in the naive case we could render out a cube map and then convert that to equirectangular via sampling, but there'd probably be artifacts at the edges of the cube from splatting approximations.

**Comment by kerrj** [comment]
Implementing other camera models for gaussian splatting is a bit nontrivial since it doesn't support rendering with arbitrary rays. One workaround could be to render small patches and stitch them together, or implement raybundle rendering with gaussians by rendering single pixels at a time (this would be much slower than rendering full images). I'm not sure the right way to implement this, open to ideas/PRs!
